FBKVOController/FBKVOController.m

```diff
@@ -275,14 +298,22 @@ - (void)unobserve:(id)object info:(_FBKVOInfo *)info
   if (nil == info) {
     return;
   }
-  
+
   // unregister info
   OSSpinLockLock(&_lock);
   [_infos removeObject:info];
   OSSpinLockUnlock(&_lock);
-  
+
   // remove observer
-  [object removeObserver:self forKeyPath:info->_keyPath context:(void *)info];
+  if (info->_state == _FBKVOInfoStateObserving) {
+    [object removeObserver:self forKeyPath:info->_keyPath context:(void *)info];
+  } else {
+    // this could happen when `NSKeyValueObservingOptionInitial` is one of the NSKeyValueObservingOptions.
+    // we cannot unobserve them right now because they are not technially observers yet.
+    // instead, we set a flag so they can be unobserved later, after the `addObserver` call returns.
+    info->_state = _FBKVOInfoStateNotObserving;
```

Comment:
You probably want to set this always, regardless of whether you are unobserving via KVO API or not.
This will prevent double calls to `removeObserver:`.